procmgr/main.cpp

```diff
@@ -2085,16 +2145,21 @@ void pingDeviceThread()
                                     Configuration config;
 									log.writeLog(__LINE__, "*** module is down: " + moduleName, LOG_TYPE_CRITICAL);
 
+                                    processManager.reinitProcessType("cpimport");
+
+                                    // halt the dbrm
+                                    oam.dbrmctl("halt");
+                                    log.writeLog(__LINE__, "'dbrmctl halt' done", LOG_TYPE_DEBUG);
+
                                     //set query system state not ready
                                     processManager.setQuerySystemState(false);
 
                                     processManager.setSystemState(oam::BUSY_INIT);
 
-                                    processManager.reinitProcessType("cpimport");
+                                    //call dbrm control
+                                    oam.dbrmctl("reload");
```

Comment:
I see a few lines later there is a disableModule call.  Has the config file been updated at this point?

Reply:
This found to be needed when a failover occurs during cpimport to keep dbrm from deadlocking. it shouldn't be an issue on failovers without cpimport, it may hang a minute while the downed node timesout trying to reconnect. and continue on with failover processes.